src/Stratis.Bitcoin/Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -58,106 +85,112 @@ private class CacheItem
         /// <summary>Default maximum number of transactions in the cache.</summary>
         public const int CacheMaxItemsDefault = 100000;
 
-        /// <summary>Length of the coinview cache flushing interval in seconds.</summary>
-        /// <seealso cref="lastCacheFlushTime"/>
-        public const int CacheFlushTimeIntervalSeconds = 60;
+        /// <summary>The current batch size in bytes.</summary>
+        private long currentBatchSizeBytes;
+
+        /// <summary>Maximum interval between saving batches.</summary>
+        /// <remarks>Interval value is a prime number that wasn't used as an interval in any other component. That prevents having CPU consumption spikes.</remarks>
+        private const int BatchMaxSaveIntervalSeconds = 41;
+
+        /// <summary>Maximum number of bytes the batch can hold until the rewind data items are stored to the disk.</summary>
+        internal const int BatchThresholdSizeBytes = 5 * 1000 * 1000;
 
         /// <summary>Instance logger.</summary>
         private readonly ILogger logger;
 
+        /// <summary>Queue which contains rewind data items that should be saved to the database.</summary>
+        private readonly AsyncQueue<QueuedRewindData> rewindDataQueue;
+
+        /// <summary>Batch of rewind data items which should be saved in the database.</summary>
+        /// <remarks>Write access should be protected by <see cref="getBlockLock"/>.</remarks>
+        private readonly List<QueuedRewindData> rewindDataBatch;
+
+        /// <summary>Task that runs <see cref="DequeueRewindDataContinuouslyAsync"/>.</summary>
+        private Task dequeueLoopTask;
+
+        /// <summary>Protects the batch from being modifying while <see cref="GetTipHashAsync"/> method is using the batch.</summary>
+        private readonly object getBlockLock;
```

Comment:
lock is not needed here. 

In BSQ it's needed because we access batch when we get the block- if it's not in batch we get it from DB. 
In case of coinview we never have to access rewind data from the outside so the lock is not needed